demos/connector/neo4j/directed-graphsage-on-cora-neo4j-example.ipynb

```diff
@@ -0,0 +1,1158 @@
+    "At this point, although we do not store all edges, it is still necessary to put all nodes in NetworkX object."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "Gnx = nx.DiGraph()\n",
+    "Gnx.add_nodes_from(node_data.index, label=\"paper\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "labels = np.array(node_data[\"subject\"])\n",
+    "\n",
+    "target_encoding = preprocessing.OneHotEncoder(sparse=False, categories=\"auto\")\n",
+    "targets = target_encoding.fit_transform(labels.reshape(-1, 1))\n",
+    "targets = pd.DataFrame(targets, index=node_data.index)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We aim to train a graph-ML model that will predict the \"subject\" attribute on the nodes. These subjects are one of 7 categories:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "{'Case_Based',\n",
+       " 'Genetic_Algorithms',\n",
+       " 'Neural_Networks',\n",
+       " 'Probabilistic_Methods',\n",
+       " 'Reinforcement_Learning',\n",
+       " 'Rule_Learning',\n",
+       " 'Theory'}"
+      ]
+     },
+     "execution_count": 8,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "set(node_data[\"subject\"])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Splitting the data"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "For machine learning we want to take a subset of the nodes for training, and use the rest for testing. We'll use scikit-learn again to do this"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "train_data, test_data = model_selection.train_test_split(\n",
+    "    node_data, train_size=0.1, test_size=None, stratify=node_data[\"subject\"]\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Note using stratified sampling gives the following counts:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "Counter({'Genetic_Algorithms': 42,\n",
+       "         'Neural_Networks': 81,\n",
+       "         'Theory': 35,\n",
+       "         'Probabilistic_Methods': 42,\n",
+       "         'Rule_Learning': 18,\n",
+       "         'Case_Based': 30,\n",
+       "         'Reinforcement_Learning': 22})"
+      ]
+     },
+     "execution_count": 10,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "from collections import Counter\n",
+    "\n",
+    "Counter(train_data[\"subject\"])"
```

Comment:
I think we could consider removing a lot of this exploration stuff from this notebook, and just refer the user to the other directed/undirected graphsage notebooks. If we removed:

- this cell [10] (along with the explanation "Note using stratified sampling gives the following counts:"  and "The training set has class imbalance ...")
- `train_data[["subject"]]` [11]
- `print(train_targets)` end of [12]

The notebook would be a bit simpler, and we could keep the focus on the Neo4J code. What do you think?

Reply:
Yeah! I will delete this out right now!